pkg/auth/config/source/local_file.go

```diff
@@ -55,5 +56,16 @@ func (s *LocalFile) Shutdown() error {
 }
 
 func (s *LocalFile) ProvideConfig(ctx context.Context, r *http.Request) (*config.Config, error) {
-	return s.config, nil
+	if s.serverConfig.DevMode {
+		// Accept all hosts under development mode
+		return s.config, nil
+	}
+
+	host := httputil.GetHost(r, s.serverConfig.TrustProxy)
+	for _, h := range s.config.AppConfig.HTTP.Hosts {
+		if h == host {
+			return s.config, nil
+		}
+	}
+	return nil, ErrAppNotFound
```

Comment:
The error message is not very helpful. Maybe generate an error at runtime

```golang
fmt.Errorf("host `%s` is not listed as valid host", host) 
```

Reply:
Most likely this error message would be written to HTTP response, and I feel uneasy to echo untrusted input in HTTP response. I suppose I'll change it to `fmt.Errorf("request host is not valid: %w", ErrAppNotFound)`, so caller can use `errors.Is(err, ErrAppNotFound)` to determine correct status code.

Reply:
It would be more helpful if we include how we get the host. Of course, the getHost function have to report the header name as well.

```golang
fmt.Errorf("host found in `%s` is invalid", headerName)
```

Reply:
I'm not sure if it would be helpful, since the host can be found in the headers rather easily?

Reply:
They may not know they have to set `TRUST_PROXY` to `true`. This error message may give the developer some hints.

I agree we should not write untrusted string to response. Maybe we should instead log a warning containing the header name and the value. It would be most helpful.

Reply:
Sure. WARN level would be noisy, since it can be frequently hit on Internet by bad actors. I suppose we can log down the request host value as DEBUG level, since DEBUG level should be off at production.

I'll add a TODO to update this after refactoring the DI mechanism (we need a logger in there).